src/nvim/mark.c

```diff
@@ -887,6 +887,18 @@ void ex_changes(exarg_T *eap)
  */
 void mark_adjust(linenr_T line1, linenr_T line2, long amount, long amount_after)
 {
+  mark_adjust_internal(line1, line2, amount, amount_after, true);
+}
+
+void mark_adjust_nofold(linenr_T line1, linenr_T line2, long amount,
```

Comment:
The doc for `mark_adjust` mentions:

> Must be called before changed_*(), appended_lines() or deleted_lines().

and it's a common pattern throughout the code. So the fact that `mark_adjust_nofold` _doesn't_ work like that could be surprising/confusing. To start with, the doc for `mark_adjust_nofold` should emphasize this situation, and I guess also mention the relationship to `foldSwapRange()`.